cnf-tests/testsuites/pkg/features/features.go

```diff
@@ -0,0 +1,222 @@
+}
+
+type DPDKFixture struct {
+}
+
+func (p *DPDKFixture) Setup() error {
+	return namespaces.Create(dpdkNS, testclient.Client)
+}
+
+func (p *DPDKFixture) Cleanup() error {
+	return namespaces.Delete(dpdkNS, testclient.Client)
+}
+
+type GatekeeperFixture struct {
+}
+
+func (p *GatekeeperFixture) Setup() error {
+	return namespaces.Create(gatekeeperNS, testclient.Client)
+}
+
+func (p *GatekeeperFixture) Cleanup() error {
+	return namespaces.Delete(gatekeeperNS, testclient.Client)
+}
+
+type SROFixture struct {
+}
+
+func (p *SROFixture) Setup() error {
+	return namespaces.Create(sroNS, testclient.Client)
+}
+
+func (p *SROFixture) Cleanup() error {
+	sro.Clean()
+	return namespaces.Delete(sroNS, testclient.Client)
+}
+
+type PTPFixture struct {
+}
+
+func (p *PTPFixture) Setup() error {
+	return nil
+}
+
+func (p *PTPFixture) Cleanup() error {
+	ptp.Clean()
+	return ptpClean.All()
+}
+
+type SriovFixture struct {
+}
+
+func (p *SriovFixture) Setup() error {
+	return nil
+}
+
+func (p *SriovFixture) Cleanup() error {
+	sriovClean.All()
+
+	err := namespaces.Delete(tuningcniSriovNS, testclient.Client)
+	if err != nil {
+		return err
+	}
+
+	return namespaces.Delete(sriovNS, testclient.Client)
+}
+
+type SCTPFixture struct {
+}
+
+func (p *SCTPFixture) Setup() error {
+	return nil
+}
+
+func (p *SCTPFixture) Cleanup() error {
+	err := namespaces.Clean("default", "testsctp-", testclient.Client)
+	if err != nil {
+		return fmt.Errorf("failed to clean 'default' namespace 'testsctp' prefix")
+	}
+
+	return namespaces.Delete(sctpNS, testclient.Client)
+}
+
+type XTU32Fixture struct {
+}
+
+func (p *XTU32Fixture) Setup() error {
+	return nil
+}
+
+func (p *XTU32Fixture) Cleanup() error {
+	return namespaces.Delete(xtu32NS, testclient.Client)
+}
+
+type VRFFixture struct {
+}
+
+func (p *VRFFixture) Setup() error {
+	return nil
+}
+
+func (p *VRFFixture) Cleanup() error {
+	return namespaces.Delete(vrfNS, testclient.Client)
+}
+
+type OVSQOSFixture struct {
+}
+
+func (p *OVSQOSFixture) Setup() error {
+	return nil
+}
+
+func (p *OVSQOSFixture) Cleanup() error {
+	return namespaces.Delete(ovsqosNS, testclient.Client)
+}
+
+type NumaresourcesFixture struct {
+}
+
+func (p *NumaresourcesFixture) Setup() error {
+	// note this intentionally does NOT set the infra we depends on the configsuite for this
+	return numaserialconf.SetupFixture()
+	// note we do NOT CHECK for error to have occurred - intentionally.
+	// Among other things, this function gets few NUMA resources-specific objects.
+	// In case we do NOT have the NUMA resources CRDs deployed, the setup will fail.
+	// But we cannot know until we run the tests, so we handle this in the tests themselves.
+	// This will be improved in future releases of the numaresources operator.
+}
+
+func (p *NumaresourcesFixture) Cleanup() error {
+	numaserialconf.Teardown()
+	return nil
+}
+
+type FECFixture struct {
+}
+
+func (p *FECFixture) Setup() error {
+	return nil
```

Comment:
why we don't create the namespace here?

Reply:
we weren't creating it before. i didn't want to change the existing setup https://github.com/openshift-kni/cnf-features-deploy/blob/57a05afbe836e9210ad638f6abe65f72b569148a/cnf-tests/testsuites/e2esuite/test_suite_test.go#L103